### PR TITLE
Add support for PR numbers

### LINKED ISSUE #12: [RFE] Support PR Numbers
Add support for PR numbers as the `ref` in any of the environment variables.

Example: `TEST_REPO=manageiq#1234 bundle exec rake`

### PR DESCRIPTION
**Pull request by agrare**
Allow for repositories to be specified by pull-request number with the 
standard repo#prnumber format

Fixes https://github.com/Fryguy/manageiq-cross_repo/issues/12

### CODE REVIEW

File: lib/manageiq/cross_repo/repository.rb [hunk 1]
```diff
@@ -46,14 +46,26 @@ def parse_identifier(identifier, server)
...
         org, repo = name.split("/")
         repo, org = org, "ManageIQ" if repo.nil?
 
         url  = File.join(server, org, repo)
-        sha  = ref.to_s.match?(/^\h+$/) ? ref : `git ls-remote #{url} #{ref}`.split("\t").first
+
+        sha = if pr
+          `git ls-remote #{url} refs/pull/#{pr}/head`.split("\t").first
+        else
+          ref.to_s.match?(/^\h+$/) ? ref : `git ls-remote #{url} #{ref}`.split("\t").first
```
Comment (Fryguy) [inline lib/manageiq/cross_repo/repository.rb]: Might want to switch this to being part of the if/else directly... like

```ruby
if pr
  git ls...
elsif ref.to_s.match?
  ref
else
  git ls...
end
```

Comment (agrare) [reply lib/manageiq/cross_repo/repository.rb]: Funny you mention that https://github.com/Fryguy/manageiq-cross_repo/pull/15/files#diff-32172a7018d252026b1910784c3806e4R67

File: README.md [hunk 2]
```diff
@@ -6,6 +6,21 @@
...
   bundle exec rake
 
+  There are three environment variables that control how the tests are run.
+
+  TEST_REPO: This is the repository which will be tested.
+  CORE_REPO: If TEST_REPO is a plugin and you need to specify a different core branch
+  GEM_REPOS: Optional other plugins/gems which are needed to run the tests
+
+  All of these support the following formats:
+  Remote: [Org/]Repostory[@Ref][#PR]
```
Comment (Fryguy) [inline README.md]: `[Org/]Repostory[@Ref|#PR]`  (Using the | to denote an "or")

Comment (agrare) [reply README.md]: :+1:

File: README.md [hunk 3]
```diff
@@ -6,6 +6,21 @@
 Usage:
   bundle exec rake
 
+  There are three environment variables that control how the tests are run.
+
+  TEST_REPO: This is the repository which will be tested.
+  CORE_REPO: If TEST_REPO is a plugin and you need to specify a different core branch
+  GEM_REPOS: Optional other plugins/gems which are needed to run the tests
```
Comment (Fryguy) [inline README.md]: Maybe mention that it's a comma-separated list?

File: README.md [hunk 4]
```diff
@@ -6,6 +6,21 @@
...
+
+  All of these support the following formats:
+  Remote: [Org/]Repostory[@Ref|#PR]
+   Org:        Optional, defaults to ManageIQ
+   Repository: Required, the name of the repository
+   @Ref:       Optional, defaults to master.  Can be either a branch name or a commit SHA.  Mutually exclusive with #pr
```
Comment (Fryguy) [inline README.md]: `Optional, defaults to master.  Can be any ref (branch, tag, or SHA).  Mutually exclusive with #PR`

File: README.md [hunk 5]
```diff
@@ -6,6 +6,21 @@
...
+  TEST_REPO: This is the repository which will be tested.
+  CORE_REPO: If TEST_REPO is a plugin and you need to specify a different core branch
+  GEM_REPOS: Optional a comma-separated-list of other plugins/gems which are needed to run the tests
+
+  All of these support the following formats:
+  Remote: [Org/]Repostory[@Ref|#PR]
+   Org:        Optional, defaults to ManageIQ
+   Repository: Required, the name of the repository
+   @Ref:       Optional, defaults to master. Can be any ref (branch, tag, or SHA). Mutually exclusive with #PR
+   #PR:        Optional, references a pull-request number.  Mutually exclusive with @ref
```
Comment (Fryguy) [inline README.md]: `@Ref` (Consistency on the capitalization)

File: README.md (hunk 4 above)
Comment (agrare) [reply README.md]: Done

File: README.md (hunk 5 above)
Comment (agrare) [reply README.md]: Done

File: README.md (hunk 3 above)
Comment (agrare) [reply README.md]: :+1: done

File: Rakefile [hunk 6]
```diff
@@ -6,6 +6,21 @@ def usage
...
+       Org:        Optional, defaults to ManageIQ
+       Repository: Required, the name of the repository
+       @Ref:       Optional, defaults to master. Can be any ref (branch, tag, or SHA). Mutually exclusive with #PR
+       #PR:        Optional, references a pull-request number.  Mutually exclusive with @Ref
+
+      Local: Either a fully qualified path or a relative path e.g. /path/to/repo, ~/relative/to/home, ../relative/to/current/dir
+
     Examples:
     # Test a specific plugin against ManageIQ master
     TEST_REPO=manageiq-ui-classic bundle exec rake
```
Comment (Fryguy) [inline Rakefile]: Minor, but can we indent these example 2 spaces under the Examples: header?  (maybe separate PR)

File: Rakefile [hunk 7]
```diff
@@ -6,6 +6,21 @@ def usage
     Usage:
       bundle exec rake
 
+      There are three environment variables that control how the tests are run.
+
+      TEST_REPO: This is the repository which will be tested.
```
Comment (Fryguy) [inline Rakefile]: Can we say this is required?

File: Rakefile [hunk 8]
```diff
@@ -6,6 +6,21 @@ def usage
...
+      CORE_REPO: If TEST_REPO is a plugin and you need to specify a different core branch
+      GEM_REPOS: Optional a comma-separated-list of other plugins/gems which are needed to run the tests
```
Comment (Fryguy) [inline Rakefile]: `Optional, a comma-separated list of ...`

File: Rakefile (hunk 6 above)
Comment (agrare) [reply Rakefile]: Yes, done

File: Rakefile (hunk 7 above)
Comment (agrare) [reply Rakefile]: Done, `Required, This is...`

File: Rakefile (hunk 8 above)
Comment (agrare) [reply Rakefile]: :+1: done

**Pull request merged**